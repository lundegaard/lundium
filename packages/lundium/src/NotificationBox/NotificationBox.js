import React from 'react';
import PropTypes from 'prop-types';
import { cx } from 'ramda-extension';

import Grid from '../Grid';
import Card from '../Card';
import Icon from '../Icon';
import Text from '../Text';

const iconPropsByKind = {
	info: { type: 'info', color: 'blue' },
	success: { type: 'success', color: 'success' },
	warning: { type: 'error', color: 'danger' },
};

const NotificationBox = ({ children, kind = 'info', className, ...rest }) => (
	<Card
		borderRadius
		boxShadow={['none', 'sm']}
		className={cx(
			'notification-box',
			{ [`notification-box--kind-${kind}`]: !!kind },
			className,
		)}
		py={{ xs: 2, md: 3 }}
		px={{ xs: 3, md: 4 }}
		{...rest}
	>
		<Grid flex>
			<Icon {...iconPropsByKind[kind]} size="18" mr={3} />
			<Text mb={0}>{children}</Text>
		</Grid>
	</Card>
);

NotificationBox.propTypes = {
	/** Children to be rendered in the main container. */
	children: PropTypes.node,
	/** Class name for root. */
	className: PropTypes.string,
	kind: PropTypes.oneOf(['info', 'warning', 'success']),
};

export default NotificationBox;
