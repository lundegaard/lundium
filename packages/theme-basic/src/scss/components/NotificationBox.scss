@import '../core';

.notification-box {
	border: 1px solid transparent;

	&--kind-info {
		background-color: $notification-box-info-bg;
		border-color: $notification-box-info-border-color;
	}

	&--kind-success {
		background-color: $notification-box-success-bg;
		border-color: $notification-box-success-border-color;
	}

	&--kind-warning {
		background-color: $notification-box-warning-bg;
		border-color: $notification-box-warning-border-color;
	}
}
