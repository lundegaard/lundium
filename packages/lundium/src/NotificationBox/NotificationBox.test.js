import React from 'react';
import { shallow } from 'enzyme';

import NotificationBox from './NotificationBox';

describe('<NotificationBox />', () => {
	const requiredProps = {};

	it('matches snapshot when shallowly rendered', () => {
		const wrapper = shallow(<NotificationBox {...requiredProps} />);
		expect(wrapper).toMatchSnapshot();
	});
});
