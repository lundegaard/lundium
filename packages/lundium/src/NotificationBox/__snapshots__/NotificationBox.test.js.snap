// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NotificationBox /> matches snapshot when shallowly rendered 1`] = `
<Card
  borderRadius={true}
  boxShadow={
    Array [
      "none",
      "sm",
    ]
  }
  className="notification-box notification-box--kind-info"
  px={
    Object {
      "md": 4,
      "xs": 3,
    }
  }
  py={
    Object {
      "md": 3,
      "xs": 2,
    }
  }
>
  <forwardRef(Grid)
    flex={true}
  >
    <Icon
      color="blue"
      mr={3}
      size="18"
      type="info"
    />
    <forwardRef(Text)
      mb={0}
    />
  </forwardRef(Grid)>
</Card>
`;
